compair/api/assignment.py

```diff
@@ -93,13 +96,15 @@ def get(self, course_uuid, assignment_uuid):
     def post(self, course_uuid, assignment_uuid):
         course = Course.get_active_by_uuid_or_404(course_uuid)
         assignment = Assignment.get_active_by_uuid_or_404(assignment_uuid)
-        require(EDIT, assignment)
+        require(EDIT, assignment,
+            title="Assignment Update Failed",
+            message="You do not have permission to update this assignment since you are not its instructor.")
 
         params = existing_assignment_parser.parse_args()
 
         # make sure the assignment id in the url and the id matches
         if params['id'] != assignment_uuid:
-            return {"error": "Assignment id does not match URL."}, 400
+            abort(400, title="Assignment Update Failed", message="Assignment id does not match URL.")
```

Comment:
Yes, these and the invalid types error are edge cases that should only happen when manually editing form data or similar actions.